### PR TITLE
vm fix for bitwise signed ints

### LINKED ISSUE #10482: 32 bit signed xor broken on VM
the bit `xor` operator doesn't work properly in the vm on signed integer types.

The cause of the bug is, on the VM negatives numbers are represented in higher precision. So a -1 has 64 bits set to 1, even if it represents a 32 bit -1. In combination with xor and a positive number, these upper bits are set to 0. If only the actual 32 important bits would be taken care of everything would seem fine, but sadly the CPU thinks these are now positive numbers.

### Example

```nim
import strutils

static:  # only the visible part is printed
  let x = -1'i32
  let y = 1'i32
  echo toHex(x)            #   FFFFFFFF
  echo toHex(y)            #   00000001
  echo toHex(x xor y)      #   FFFFFFFE
  echo toHex(-2'i32)       #   FFFFFFFE  <- seems right doesn't it ?
  echo int64(x xor y)        #   4294967294  <- something is wrong here
  echo toHex(int64(x xor y)) #   00000000FFFFFFFE  <-\
  echo toHex(int64(-2))      #   FFFFFFFFFFFFFFFE  <---- the difference is in the hidden part
```

### PR DESCRIPTION
**Pull request by cooldome**
fixes #10482

I have found that I can't sort in vm when I compile with `--cpu:i386`.
Sort is using xor
https://github.com/nim-lang/Nim/blob/acac107fe7fbef2b55cef3264e9e5e10e1f69ea3/lib/pure/algorithm.nim#L65

### CODE REVIEW

**Comment by mratsim** [comment]
This seems to cause issues for Javascript with GCC:
  - Windows (mingw) and Linux fails.
  - OSX (Clang) passes.

Or are the failures unrelated?

File: compiler/vmgen.nim [hunk 1]
```diff
@@ -986,11 +986,11 @@ proc genMagic(c: PCtx; n: PNode; dest: var TDest; m: TMagic) =
...
 
-  of mShlI: genBinaryABCnarrowU(c, n, dest, opcShlInt)
+  of mShlI: genBinaryABCnarrow(c, n, dest, opcShlInt)
   of mAshrI: genBinaryABCnarrow(c, n, dest, opcAshrInt)
-  of mBitandI: genBinaryABCnarrowU(c, n, dest, opcBitandInt)
-  of mBitorI: genBinaryABCnarrowU(c, n, dest, opcBitorInt)
-  of mBitxorI: genBinaryABCnarrowU(c, n, dest, opcBitxorInt)
+  of mBitandI: genBinaryABCnarrow(c, n, dest, opcBitandInt)
+  of mBitorI: genBinaryABCnarrow(c, n, dest, opcBitorInt)
+  of mBitxorI: genBinaryABCnarrow(c, n, dest, opcBitxorInt)
```
Comment (krux02) [inline compiler/vmgen.nim]: you can leave out the narrowing step entirely. `and`, `or` and `xor` can only reduce the total amount of set bits.

**Comment by krux02** [comment]
@mratsim what does Javascript have to do with Gcc?

File: compiler/vmgen.nim (hunk 1 above)
Comment (cooldome) [reply compiler/vmgen.nim]: done

**Pull request merged**